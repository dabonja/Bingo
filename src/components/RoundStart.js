import React from 'react';

const RoundStart =(props)=>{
  const {onClick, disabled,currentNumber, round,color} = props;

  const clr = color == ''  ? 'none' : 'block';
    return(
      <div  className="item-a" style={{color:'rgba(244, 102, 255, 1)'}} data-testid="roundStart">
          <h1 className="display-4">Bingo</h1>
          <p className="lead">Round: {round}</p>
            <button onClick={onClick} disabled={disabled} type="button" className="btn btn-danger btn-lg" >Start</button>
            <div className="crr">
            <div className="outer-ball" style={{width: '100px', height: '100px',position:'relative', top:'8rem',display: clr,backgroundColor: color,boxShadow: 'rgba(0, 0, 0, 0.56) 0px 22px 70px 4px'}}>
              <div className="inner-ball" style={{color: 'black',width: '62px', height: '62px',fontSize:'36px'}}>
                {currentNumber}
              </div>
            </div>
            </div>


      </div>
    )
}

export default RoundStart
