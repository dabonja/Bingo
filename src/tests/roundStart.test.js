import {render,screen,cleanup} from '@testing-library/react';
import RoundStart from '../components/RoundStart';
import { configure } from "enzyme";
import { shallow } from 'enzyme';
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });
import renderer from 'react-test-renderer';

//onClick

afterEach(()=>{
cleanup();
});


const num = 23;
const rNum = 2;
const color = "#00FF00";
const disabled = true;

describe('renders components correctly', ()=>{
  it("renders RoundStart component without crashing",()=>{
    const tree = renderer.create(<RoundStart />).toJSON();
    expect(tree).toMatchSnapshot();
  })


  it('renders header correctly',()=>{
      const wrapper = shallow(<RoundStart/>);
      const header =  <h1 className="display-4">Bingo</h1>;
      expect(wrapper.contains(header)).toEqual(true);

  })
})



test('round start should contain current number',()=>{
  render(<RoundStart currentNumber={num}/>);
  const start = screen.getByTestId("roundStart");

  expect(start).toBeInTheDocument();
  expect(num).toBe(23);
  expect(rNum).toBe(2);
  expect(color).toBe("#00FF00");
  expect(disabled).toBeTruthy();
  expect(disabled).not.toBeFalsy();

})
