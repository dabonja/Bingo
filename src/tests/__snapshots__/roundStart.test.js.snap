// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders RoundStart component without crashing 1`] = `
<div
  className="item-a"
  data-testid="roundStart"
  style={
    Object {
      "color": "rgba(244, 102, 255, 1)",
    }
  }
>
  <h1
    className="display-4"
  >
    Bingo
  </h1>
  <p
    className="lead"
  >
    Round: 
  </p>
  <button
    className="btn btn-danger btn-lg"
    type="button"
  >
    Start
  </button>
  <div
    className="outer-ball"
    style={
      Object {
        "backgroundColor": undefined,
        "boxShadow": "rgba(0, 0, 0, 0.56) 0px 22px 70px 4px",
        "display": "block",
        "height": "100px",
        "position": "relative",
        "top": "18%",
        "width": "100px",
      }
    }
  >
    <div
      className="inner-ball"
      style={
        Object {
          "color": "black",
          "fontSize": "36px",
          "height": "62px",
          "width": "62px",
        }
      }
    />
  </div>
</div>
`;

exports[`renders components correctly renders RoundStart component without crashing 1`] = `
<div
  className="item-a"
  data-testid="roundStart"
  style={
    Object {
      "color": "rgba(244, 102, 255, 1)",
    }
  }
>
  <h1
    className="display-4"
  >
    Bingo
  </h1>
  <p
    className="lead"
  >
    Round: 
  </p>
  <button
    className="btn btn-danger btn-lg"
    type="button"
  >
    Start
  </button>
  <div
    className="outer-ball"
    style={
      Object {
        "backgroundColor": undefined,
        "boxShadow": "rgba(0, 0, 0, 0.56) 0px 22px 70px 4px",
        "display": "block",
        "height": "100px",
        "position": "relative",
        "top": "18%",
        "width": "100px",
      }
    }
  >
    <div
      className="inner-ball"
      style={
        Object {
          "color": "black",
          "fontSize": "36px",
          "height": "62px",
          "width": "62px",
        }
      }
    />
  </div>
</div>
`;
